coalib/bearlib/aspects/Smell.py

```diff
@@ -101,12 +100,57 @@ def func(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, z):
         parameters (maybe by creating more functions or using libraries).
         """
 
+
+@MethodSmell.subaspect
+class LongMethod:
+    """
+    This aspect checks for methods and functions with a large number
+    of lines.
+    """
+    class docs:
+        example = """
+            # Here is a method with one line of code
+            def func():
+                print('coala is always written with lowercase `c`)
+        """
+        example_language = 'python'
+        importance_reason = """
+        The longer a method or function is, the harder it becomes to
+        understand and maintain it. Also long methods easily introduce
+        unwanted duplicate code to our codebase.
+        """
+        fix_suggestions = """
+        Extracting methods or functions is a fix for this.
+        """
     max_method_length = Taste[int](
         'Represents the max number of lines for a method or a function\'s'
         'definition.',
-        (40,), default=40)
+        (5, 40), default=40)
+
+
+@MethodSmell.subaspect
+class LongParameterList:
+    """
+    This aspect checks for methods and functions with a large number of
+    paramters.
+    """
+    class docs:
+        example = """
+        def func(a, b, c, d, e, f, g, h, i, j, k):
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Refactoring such methods and functions may reveal some duplicated
+        code, make the code shorter, easy to read and maintain
+        """
+        fix_suggestions = """
+        Usually passing objects to functions or methods instead of passing
+        their attributes, if possible grouping some paramaters into a single
+        object, splitting up methods... can fix this.
```

Comment:
he fixed it in a later commit. i think that change should be here, not in that commit. why introduce a bug just to fix it a commit later?